File: app/lib/domain/models/chat_message.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'chat_message.freezed.dart';
part 'chat_message.g.dart';

enum ChatRole { system, user, assistant }

@freezed
class ChatMessage with _$ChatMessage {
  const factory ChatMessage({
    required int planId,
    required ChatRole role,
    required String content,
    required DateTime createdAt,
  }) = _ChatMessage;

  factory ChatMessage.fromJson(Map<String, dynamic> json) =>
      _$ChatMessageFromJson(json);
}

File: app/android/app/src/main/java/com/gymcoach/gym_coach/domain/model/ChatMessage.kt
package com.gymcoach.gym_coach.domain.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ChatMessage(
    @SerialName("plan_id") val planId: Int,
    val role: ChatRole,
    val content: String,
    @SerialName("created_at") val createdAt: Long
)
